cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -1069,6 +1104,12 @@ impl ManagedEnvironment {
         local: &CoreEnvironment,
         remote: &CoreEnvironment,
     ) -> Result<bool, ManagedEnvironmentError> {
+        let local_lockfile_bytes = local.existing_lockfile_contents()?;
+        let remote_lockfile_bytes = remote.existing_lockfile_contents()?;
+        if local_lockfile_bytes != remote_lockfile_bytes {
+            return Ok(false);
+        }
+
```

Comment:
@ysndr I can't remember is there a reason we don't want to check the lockfile in `validate_checkout`?

Reply:
I dont think so, its not that we really check lockfiles anywhere tho ^^
there might be issues with pulling incompatible/older lockfiles, newer ones as well but we arent really forward compatible anyway